src/Controls/src/BindingSourceGen/PathParser.cs

```diff
@@ -68,26 +64,19 @@ internal static bool ParsePath(CSharpSyntaxNode? expressionSyntax, bool enabledN
 
             var defaultMemberName = "Item"; // TODO we need to check the value of the `[DefaultMemberName]` attribute on the member type
             IPathPart part = new IndexAccess(defaultMemberName, indexValue);
-            if (isNodeNullable || BindingGenerationUtilities.IsTypeNullable(typeInfo, enabledNullable))
-            {
-                part = new ConditionalAccess(part);
-            }
             parts.Add(part);
             return true;
         }
         else if (expressionSyntax is ConditionalAccessExpressionSyntax conditionalAccess)
         {
-            return ParsePath(conditionalAccess.Expression, enabledNullable, context, parts, isNodeNullable: true) &&
+            return ParsePath(conditionalAccess.Expression, enabledNullable, context, parts) &&
             ParsePath(conditionalAccess.WhenNotNull, enabledNullable, context, parts);
         }
         else if (expressionSyntax is MemberBindingExpressionSyntax memberBinding)
         {
             var member = memberBinding.Name.Identifier.Text;
             IPathPart part = new MemberAccess(member);
-            if (isNodeNullable)
-            {
-                part = new ConditionalAccess(part);
-            }
+            part = new ConditionalAccess(part);
```

Comment:
This confuses me a little. What exactly is `MemberBindingExpressionSyntax`? Should the `part` always be conditional in this case?

Reply:
This should represent the '.?Property' part of ConditionalAccess.